src/math/trig.c

```diff
@@ -270,6 +330,22 @@ int32_t is_scalar_cordic_asin(int32_t sinvalue, int16_t numiters)
  *		  struct  cordic_cmpx
  * Return Type	: none
  */
+#ifdef CORDIC_HIFI3
+void cmpx_cexp(ae_int32 sign, ae_int32 b_yn, ae_int32 xn, cordic_cfg type, struct cordic_cmpx *cexp)
+{
+	ae_int32 s;
+
+	cexp->re = sign * xn;
+	cexp->im = sign * b_yn;
```

Comment:
This might be possible as dual multiply

Reply:
Seems feasible only for loop since SIMD setup to ae_f32x2 and return to complex number placeholder struct takes more cycles.